app/routers/match_router.py

```diff
@@ -0,0 +1,54 @@
+from fastapi import APIRouter
+
+from app.data import MongoDB
+from app.schema import MatchQuery, MatchUpdate
+
+Router = APIRouter(
+    tags=["Matching Operations"],
+)
+
+Router.db = MongoDB()
+
+
+@Router.post("/create/match")
+async def create_match(data: MatchUpdate):
+    """Insert a new mentor/mentee pairing into the collection
+    <pre><code>
+    @param data: JSON[MatchUpdate]
+    @return JSON[Boolean] - success or failure of update</pre></code>"""
+    data = data.dict()
+    return {"result": Router.db.upsert_to_set_array("Matches",
+                                                    {"mentor_id": data["mentor_id"]},
+                                                    {"mentee_ids": data["mentee_id"]})}
+
+
+@Router.post("/delete/match")
+async def delete_match(data: MatchUpdate):
+    """Remove a mentor/mentee pairing from the collection
+    <pre><code>
+    @param data: JSON[MatchUpdate]
+    @return JSON[Boolean] - success or failure of update</pre></code>"""
+    data = data.dict()
+    return {"result": Router.db.delete_from_array("Matches",
+                                                  {"mentor_id": data["mentor_id"]},
+                                                  {"mentee_ids": data["mentee_id"]})}
+
+
+@Router.post("/read/match")
+async def get_match(query: MatchQuery):
+    """Retrieve all matching mentor/mentee objects for a given mentee/mentor
+    <pre><code>
+    @param query: JSON[MatchQuery]
+    @return JSON[Array[Mentor]] | JSON[Array[Mentee]]</pre></code>"""
+    query = query.dict()
```

Comment:
you take the query from the input, but then later you overwrite it, not sure what was intended here.